src/tigris.ts

```diff
@@ -238,3 +237,52 @@ export class Collection<T extends TigrisCollectionType> {
  * Default instance of the Tigrisclient
  */
 export default new Tigris({serverUrl: `${process.env.TIGRIS_SERVER_URL}`});
+
+export type Filter<T extends string | number | boolean> = {
+    key: string;
+    val: T;
+}
+
+export type LogicalFilter<T extends string | number | boolean> = {
+    logicalOperator: LogicalOperator
+    filters?: Filter<T>[];
+    logicalFilters?: LogicalFilter<T>[];
+}
+
+export const Utility = {
+    filterString<T extends string | number | boolean>(filter: Filter<T>): string {
+        return JSON.stringify(this.filterJSON(filter));
+    },
+
+    filterJSON(filter: Filter<string | number | boolean>): object {
+        const obj = {};
+        obj[filter.key] = filter.val;
+        return obj;
+    },
+
+    logicalFilterString<T extends string | number | boolean>(filter: LogicalFilter<T>): string {
+        return JSON.stringify(this.logicalFilterJSON(filter));
+    },
+
+    logicalFilterJSON(filter: LogicalFilter<string | number | boolean>): object{
+        const obj = {};
```

Comment:
suggestion: We should try to prevent mutating data too much by using map and reduce as well as spread operator

Reply:
Thanks for the review. I understand there is free form object construction here. That is enclosed in this method which is non user facing. How do you suggest I construct JSON object using LogicalFilter fields?

Reply:
As discussed, I have made some changes in a new branch. One this has between merged, I will create new PR